packages/core/src/auto.ts

```diff
@@ -1816,10 +1816,15 @@ export default class Auto {
       throw this.createErrorMessage();
     }
 
+    const currentBranch = getCurrentBranch();
     const isPrerelease = this.inPrereleaseBranch();
+
+    const maintenanceMajor = this.config?.versionBranches && typeof this.config?.versionBranches === "string" && currentBranch?.startsWith(this.config.versionBranches) ? 
+      parseInt(currentBranch.replace(this.config.versionBranches, ""), 10) : null
     const [, latestTagInBranch] = await on(this.git.getLatestTagInBranch());
     const lastRelease =
       from ||
```

Comment:
From my understanding, this will prioritize the `from` value over the proper version number that you fetched with your function.

In the following line: https://github.com/intuit/auto/blob/main/packages/core/src/auto.ts#L1688, auto passes in the `latestTag` from the function `getLatestTagInBranch()` as the key `from`. 

This value still returns the incorrect tag version, and will get passed into the `getVersion` method within here: https://github.com/intuit/auto/blob/main/packages/core/src/auto.ts#L1714, then it will be prioritized over your change.